l1-contracts/src/core/interfaces/IProofCommitmentEscrow.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {DataStructures} from "../libraries/DataStructures.sol";
+
+interface IProofCommitmentEscrow {
```

Comment:
I don't think this interface will work, unless you also do a lot of the same accounting in here as you do in the rollup. We should do them in either one or the other. 

Specifically, look at the `stakeBond` and `unstakeBond` the stake gets a lot of information, some useful some not, but it knows who is bonding etc. But at the `unstake` it won't be told who is it that is unstaking which makes it weird here. 

In the likely case we have, where we are building the escrow specifically for the rollup, you can have the rollup provide enough information for the escrow to know what to do without doubling the accounting.

Reply:
I agree that one escrow per rollup is simpler- that is a deviation from what I was thinking.

Reply:
The `unstakeBond` is still not identifying who, so it still needs to store a bunch of storage on its own. I suggest making the "mock" without validation but still doing the transfers, such that you can more easily get an idea around the information that needs to be passed along for the minimum. 

Also, it is not enough to give a signature, to verify it we need to know the message etc. Either have that logic in the rollup, and then the escrow becomes just `bond(address _bonder, uint256 _amount) onlyRollup` or have it deal with the full thing.